### PR TITLE
Generate json files

### LINKED ISSUE #7: Load summary data to be used in listing views
In the initial version of the project we will be creating JSON data to be persisted within the repo. 

This task is to load the data to be used in the listing view of all projects, to be displayed in repo cards displaying the key attributes of the repo.

This task should appropriately handle loading, success and error states

### PR DESCRIPTION
**Pull request by relvao**
closes #7 

Pulls data from github and api scorecard and writes json files into the assets folder of the website.
Still needs work to adjust the schema to what is needed in the UI but will do that in a separate PR so this ins't too big

### CODE REVIEW

**Review by Puppo — CHANGES_REQUESTED** [review]

File: build-scripts/src/github.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  org: string;
+}
+
+interface GitHubClientArgs {
+  gh_priv_key: string;
+  gh_app_id: string;
+  gh_app_install_id: number;
+}
+
+export default async function* queryGitHub(args: QueryGitHubArgs & GitHubClientArgs) {
```
[the remark this answers is not in the document]
Comment (relvao) [reply build-scripts/src/github.ts:40]: I prefer to have implicit return types for no special reason than that I don't want to have to do any more work than absolutely necessary ;) 
I don't have a strong position though. I'll rewrite all fns to have explicit return types.

File: build-scripts/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  .version('1.0.0', '-v, --version')
+  .requiredOption('-p, --path <string>', 'The path where the files should be saved.')
+  .requiredOption('-o, --org <string>', 'The github organisation.')
+  .requiredOption('--gh_priv_key <string>', 'The github private key for the app.')
+  .requiredOption('--gh_app_id <string>', 'The github id of the app.')
+  .requiredOption('--gh_app_install_id <string>', 'The github id for the installed app.')
+  .parse();
+const options = program.opts();
+
+if (!fs.existsSync(options.path)) {
```
[the remark this answers is not in the document]
Comment (relvao) [reply build-scripts/src/index.ts:17]: I'll move it into a fn and make the rest of the process wait on it to be resolved.

File: build-scripts/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  .requiredOption('-o, --org <string>', 'The github organisation.')
+  .requiredOption('--gh_priv_key <string>', 'The github private key for the app.')
+  .requiredOption('--gh_app_id <string>', 'The github id of the app.')
+  .requiredOption('--gh_app_install_id <string>', 'The github id for the installed app.')
+  .parse();
+const options = program.opts();
+
+type RepoId = string;
+type DateIsoString = string;
+interface IndexData {
```
Comment (Puppo) [inline build-scripts/src/index.ts:19]: `IndexData` is a bit difficult to understand, in my opinion. Maybe something like `RepositoryInfoStatus`. I understood that it's the `index.json` content, but using a better name can help to understand what this file contains.

File: build-scripts/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  const index: IndexData[] = [] // main doc, used for searches & pagination
+  const queryGitHubArgs = {
+    org: options.org,
+    itemsPerPage: 10,
+    gh_priv_key: options.gh_priv_key,
+    gh_app_id: options.gh_app_id,
+    gh_app_install_id: options.gh_app_install_id
+  }
+
+  for await (const gitHubItem of queryGitHub(queryGitHubArgs)) {
```
Comment (Puppo) [inline build-scripts/src/index.ts:38]: This is a simple thought.
I don't know how many repositories we are and if there is a rate limit in these APIs, but using an asyncIterator, we call the API every iteration. I think this affects the execution time and the number of minutes for the GitHub Actions increases. Understanding if it's possible to resolve this issue using a `Promise.all` or a `Promise.allSettled` to "parallelise" all the requests and then manipulate the data after.
I don't want to refactor the codebase now; I just want to raise the flag on this point. This is impossible if we have a rate limit in the API, but maybe understanding this solution could be helpful. We can also go ahead with the current solution, check the time and the performance, and refactor the codebase in the future.

**Review by Puppo — DISMISSED** [review]

**Pull request merged**